### ISSUE TITLE
SMTP Templates for Specific Providers

### ISSUE THREAD

**Issue by Kryszn0** [zitadel/zitadel#7052]
My Zitadel instance does not seem to have SMTP functioning due to what looks like has been discussed in the below thread. As far as I can see, it does not appear this has been updated/resolved/tracked with further progress?

Like in the below thread, SMTP via Office365 is not working when trying to use the default config settings in the Zitadel GUI
 
///////////////////////

Yes we will have to implement it ourselves. And we will need a config or possibility on the smtp settings, where developers / users are able to specify the authentication type and also more detailed TLS settings (STARTTLS vs TLS, ...)
Best would then be to have templates for specific providers, like the IDP templates for login.

_Originally posted by @livio-a in https://github.com/zitadel/zitadel/discussions/4263#discussioncomment-4776986_

**Comment by doncicuto** [comment]
Hi @MasterKryszn, in #5503 we are trying to set a few SMTP templates that can offer some defaults for SMTP providers. 

My code is currently being reviewed but it may take some time as I suggest several changes and they have to follow Zitadel general guidelines so I may have to update the code so it's good and mature. 

I'm not familiar with Office365 SMTP settings could you provide me a link with docs containing default or usual settings for Office365 (host, port, tls, user, password). If Office365 had a trial option without credit card I may try to test my current code with that service and add it as a prospect template.

**Comment by MasterKryszn** [comment]
> Hi @MasterKryszn, in #5503 we are trying to set a few SMTP templates that can offer some defaults for SMTP providers.
> 
> My code is currently being reviewed but it may take some time as I suggest several changes and they have to follow Zitadel general guidelines so I may have to update the code so it's good and mature to be merged.
> 
> I'm not familiar with Office365 SMTP settings, could you provide me a link with docs containing default or usual settings for Office365 (host, port, tls, user, password). If Office365 had a trial option without credit card I may try to test my current code with that service and add it as a prospect template.

Not a problem, [this is Microsoft's guide for how to setup SMTP](https://learn.microsoft.com/en-us/exchange/mail-flow-best-practices/how-to-set-up-a-multifunction-device-or-application-to-send-email-using-microsoft-365-or-office-365). To clarify, it does appear as if Zitadel is already built the requires SMTP input settings to fulfill Microsofts requirements for Option 1. Included an image in this comment of Zitadel's Instance Setting page; General Category; SMTP Provider. The issue is, no matter what I put in here is saved but doesn't seem to trigger any effect when the self-hosted application is told to send out a verification email. No failures or anything in the logs. 

![smtp zitadel](https://github.com/zitadel/zitadel/assets/113187314/82432709-a162-489a-afc7-4d1200391d7d)

**Comment by doncicuto** [comment]
Hi @MasterKryszn first of all thanks a lot for the links to the docs. I'll use those settings to add a template for Office365 (Option 1) for #5503. 

Second, your issue for Option 1, I'll try to create a trial account for Office365 and try to reproduce the issue and let you know anything I found. I'm only a contributor so maybe a member of the Zitadel team can give you more info in the meantime and if you're in a hurry somebody in the Discord channel may help you (open a thread in question-bugs).

**Comment by K-J-VV** [comment]
@doncicuto any luck getting Microsoft SMTP working?

**Comment by doncicuto** [comment]
Sorry @K-J-VV some personal issues are affecting my contributions. I tried to create a trial account for Microsoft 365 but it needed a credit card so I dismissed that option. I've asked a Microsoft 365 account for a friend, but as she wasn't an admin she could not enable SMTP Auth using the Microsoft Admin Center. So far I have no way to test Option1. I've started a research about using oauth in SMTP AUTH that I'll try to test once I get #5503 closed.

**Comment by K-J-VV** [comment]
> Sorry @K-J-VV some personal issues are affecting my contributions. I tried to create a trial account for Microsoft 365 but it needed a credit card so I dismissed that option. I've asked a Microsoft 365 account for a friend, but as she wasn't an admin she could not enable SMTP Auth using the Microsoft Admin Center. So far I have no way to test Option1. I've started a research about using oauth in SMTP AUTH that I'll try to test once I get #5503 closed. 
> 
> 

Appreciate the update! Logistically, focusing on the oauth route would be best anyways as eventually Option 1 will fade away due to more modern security standards. So it would be the ideal route to take for future/longevity

**Comment by doncicuto** [comment]
Yes @K-J-VV, I totally agree. In the meantime, if you want me to test and could provide me a test mailbox for option 1 in your domain I may try to see it I can fix something about the smtp code (maybe a starttl issue) you can send me an email with test credentials.

**Comment by K-J-VV** [comment]
@doncicuto 

Thought I had replied to this, but just FYI I switched my SMTP provider and am not having issues anymore. The new provider functions by using the "old"/standard/legacy method of SMTP sending

I realized using Microsoft was giving me sporadic issues. Some hosts/programs would work, some wouldn't, some would need a different address or port for SMTP, it was complicated.

Just as a side note to mention in general: I'd still suggest someone involved in actual development of Zitadel look into this issue and figure out how to make email sending work with the new secure methods I linked above. I forsee this as a future standard/requirement, so getting it all working (and with a large prevalent provider like Microsoft) would likely be a good move.

**Comment by stebenz** [comment]
Created a summarized issue https://github.com/zitadel/zitadel/issues/8042, will move this into the backlog.

**Comment by wim07101993** [comment]
With #8042 we support XOAUTH2 which is required by microsoft/google. Closing this issue.
